observer/observer.go

```diff
@@ -202,6 +207,28 @@ func (o *Observer) IsAnyJobRunning(repository string) bool {
 	return false
 }
 
+// IsConcurrentJobsLimitReached checks if the limit of concurrent jobs by type (backup, check, etc)
+// has been reached
+func (o *Observer) IsConcurrentJobsLimitReached(jobType v1alpha1.JobType, limit int) bool {
+	if limit <= 0 {
+		return false
+	}
+	listOfJobs := o.observedJobs
```

Comment:
I'd mutex lock this assignment:

```suggestion
        o.mutex.Lock()
	listOfJobs := o.observedJobs
	o.mutex.Unlock()
```

Because writing to the map happens in a separate go routine.

The rest LGTM.